server/main.go

```diff
@@ -0,0 +1,95 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"os"
+
+	gpb "github.com/BetterGR/homework-microservice/homework_protos"
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+// homeworkService is an implementation for the Grpc homework microservice
+type homeworkServer struct {
+	//throws unimplemented exception
+	gpb.UnimplementedHomeworkServiceServer
+	logger logr.Logger
+}
+
+// logger functions
+// initializing the logger
+func newHomeworkServer(logger logr.Logger) *homeworkServer {

```

Comment:
This creates a new server. The server struct wraps a logger but that is not the entirety of its purpose.